src/host/test/ledger.cpp

```diff
@@ -92,12 +92,16 @@ size_t number_of_files_in_ledger_dir()
   return file_count;
 }
 
-size_t number_of_committed_files_in_ledger_dir()
+size_t number_of_committed_files_in_ledger_dir(bool allow_recovery = false)
 {
   size_t committed_file_count = 0;
   for (auto const& f : fs::directory_iterator(ledger_dir))
   {
-    if (is_ledger_file_committed(f.path().string()))
+    auto file_name = f.path().string();
+    if (
+      (allow_recovery && is_ledger_file_name_recovery(file_name) &&
+       file_name.find(ledger_committed_suffix) != std::string::npos) ||
```

Comment:
`.ends_with`?

Reply:
I tried this at first but that doesn't work because a committed and recovery ledger file will still end in `.recovery`, e.g. `ledger_1-2.committed.recovery`.

Reply:
Ooh, I see, they could have both suffixes, good point!